lib/src/debian.dart

```diff
@@ -0,0 +1,282 @@
+/// debian package.
+///
+/// Returns the path of created folder.
+Future<String> _createPackageDirectory(String repo, String packageName) async {
+  String debianDir = p.join(repo, packageName);
+  await Directory('$debianDir/DEBIAN').create(recursive: true);
+  await Directory('$debianDir/usr/local/bin').create(recursive: true);
+  return debianDir;
+}
+
+/// Copy all executable files listed in the map [executables] from the `build` folder
+void _copyExecutableFiles(String debianDir) {
+  final executablePath = p.join(debianDir, "usr", "local", "bin");
+  executables.value.forEach((name, path) {
+    run("cp", arguments: [
+      p.join("build", "$name.native"),
+      p.join(executablePath, name)
+    ]);
+  });
+}
+
+/// Generate the control file for the Debian package.
+void _generateControlFile(String debianDir) {
+  var controlFilePath = p.join(debianDir, "DEBIAN", "control");
+
+  String _updatedControlData = replaceFirstMappedMandatory(
+      controlData.value,
+      RegExp(r'Version: ([0-9].*)'),
+      (match) => 'Version: ${version.toString()}',
+      "Couldn't find a version field in the given CONTROL file.");
+
+  writeString(controlFilePath, _updatedControlData);
+}
+
+/// Scan for new .deb packages in the [repo] and update the `Packages` file.
+Future<void> _updatePackagesFile(String repo) async {
+  // Scan for new packages
+  String output = run("dpkg-scanpackages",
+      arguments: ["--multiversion", "."], workingDirectory: repo);
+  // Write the stdout to the file
+  writeString(p.join(repo, 'Packages'), output);
+  // Force Compress the Packages file
+  run("gzip", arguments: ["-k", "-f", "Packages"], workingDirectory: repo);
+}
+
+/// Generate the Release index for the PPA.
+Future<void> _updateReleaseFile(String repo) async {
+  String output = run("apt-ftparchive",
+      arguments: ["release", "."], workingDirectory: repo);
+  writeString(p.join(repo, 'Release'), output);
+}
+
+/// Sign the Release file with the GPG key.
+Future<void> _updateReleaseGPGFile(String repo) async {
+  run("gpg",
+      arguments: [
+        for (String arg in gpgArgs) arg,
+        "--default-key",
+        (gpgFingerprint.value),
+        "--passphrase",
+        (gpgPassphrase.value),
+        "-abs",
+        "-o",
+        "Release.gpg",
+        "Release",
+      ],
+      quiet: true,
+      workingDirectory: repo);
+}
+
+/// Update the InRelease file with the new index and keys.
+Future<void> _updateInReleaseFile(String repo) async {
+  run("gpg",
+      arguments: [
+        for (String arg in gpgArgs) arg,
+        "--default-key",
+        (gpgFingerprint.value),
+        "--passphrase",
+        (gpgPassphrase.value),
+        "--clearsign",
+        "-o",
+        "InRelease",
+        "Release",
+      ],
+      quiet: true,
+      workingDirectory: repo);
+}
+
+/// Import the private key into the GPG.
+void _importGPGPrivateKey() {
+  log("Importing the GPG Private Key");
+  writeString("._privateKey", gpgPrivateKey.value);
```

Comment:
Rather than writing this to a temporary file, can you have GPG read it from `/dev/stdin`? You may need to use the lower-level [`Process.start()`](https://api.dart.dev/stable/2.16.0/dart-io/Process/start.html).

Reply:
Like echoing the private key and pipelining the output to `gpg`?

Reply:
Essentially yes, but you'd be writing to [`Process.stdin`](https://api.dart.dev/stable/2.16.0/dart-io/Process/stdin.html) yourself.

Reply:
Ok, I'll try.

Reply:
I tried using `Process.start()`, but seems like it isn't working properly. I checked the following command works fine:
```sh
cat privateKey | gpg --allow-secret-key-import --import
```

The below piece of code never gets the exitCode and is stuck. In my opinion, it is working similar to running the command `gpg --allow-secret-key-import --import`. This command never ends.

```dart
  var process = await Process.start(
    "gpg",
    [
      ..._gpgArgs,
      "--passphrase",
      gpgPassphrase.value,
      "--import",
    ],
  );
  log('Writing STDIN for the process');
  process.stdin.write(gpgPrivateKey.value);
  var exitCode = await process.exitCode;
  log('Got the exit code');
```

Can you tell where am I going wrong?

Reply:
You need to call `process.stdin.close()` after `process.stdin.write()`. Otherwise, you'll never send the end-of-file marker that tells GPG the private key is complete.

Reply:
Oh, so the `process.stdin.close()` function sends the EOF to the process. Thanks, it worked.